### PR TITLE
[Impeller] make sure binding nullptr texture does not crash.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/158074

binding a nullptr texture should fail but not crash.

### CODE REVIEW

File: impeller/renderer/backend/metal/render_pass_mtl.mm [hunk 1]
```diff
@@ -408,6 +408,9 @@ static bool Bind(PassBindingsCacheMTL& pass,
     const ShaderMetadata& metadata,
     std::shared_ptr<const Texture> texture,
     const std::unique_ptr<const Sampler>& sampler) {
+  if (!texture) {
+    return false;
+  }
```
Comment (gaaclarke) [inline impeller/renderer/backend/metal/render_pass_mtl.mm:413]: nit: I would prefer this is an assert instead of turning BindResource to a noop.  It's harder to follow code if every procedure call is prefaced with an implicit "maybe", like "MaybeBindResource()".  New people on the team complain about this pattern making the code hard to follow.

This at least communicates when nothing has happened with the returnval and matches existing patterns though so we shouldn't hold back the PR about it.  How about adding `[[nodiscard]]` to this method to communicate at callsites that the method potentially is a noop?

**Review by gaaclarke — APPROVED** [review]

File: impeller/renderer/backend/metal/render_pass_mtl.mm (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/metal/render_pass_mtl.mm:413]: We don't want to crash the app if one of the draws is misconfigured. There is some ambiguity (due to runtime effect) as to whether this is covering up a bug in flutter or a bug in the user code. 

DCHECK would be fine, except that it doesn't stop the production crash. 

What I'd really like is to get those nullability annotations working correctly so we can correctly eliminate the nullptr closer to where it was generated.
